src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -1959,11 +1959,17 @@ internal static bool AccessingAutopropertyFromConstructor(BoundExpression receiv
             return (object)sourceProperty != null &&
                     sourceProperty.IsAutoProperty &&
                     sourceProperty.ContainingType == fromMember.ContainingType &&
-                    ((MethodSymbol)fromMember).MethodKind == (propertyIsStatic ? MethodKind.StaticConstructor
-                                                                                : MethodKind.Constructor) &&
-                   (propertyIsStatic || receiver.Kind == BoundKind.ThisReference);
+                    IsConstructorOrField(fromMember, isStatic: propertyIsStatic) &&
+                    (propertyIsStatic || receiver.Kind == BoundKind.ThisReference);
         }
 
+        private static bool IsConstructorOrField(Symbol member, bool isStatic)
```

Comment:
This is basically a part of a complicated condition refactored into a private helper and is used in a single place (could be a good case for nested methods). 
I am not able to name it much better :-). I can consider any suggestions, but as far as names I can come up with, the current one seems ok.